From ddff43913a6ebc8456208cccdf7f46ded2efdcf0 Mon Sep 17 00:00:00 2001
From: Hariharan Basuthkar <quic_hbasuthk@quicinc.com>
Date: Mon, 30 Sep 2024 15:16:13 +0530
Subject: [PATCH] hostapd: Add QCA_WLAN_VENDOR_ATTR_AFC_RESP_HW_IDX to afc
 response enum
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

In a scenario where a wiphy contains more than one 6 GHz radio,
the AFC response cannot be routed to the respective radio with the
netdev in a single wiphy architecture.

Therefore, add a new identifier ‘QCA_WLAN_VENDOR_ATTR_AFC_RESP_HW_IDX’
in the qca_wlan_vendor_attr_afc_response, which can be used to route
the AFC response to the respective radio for writing to shared memory.

Tag: 6G_REGULATORY
Patch-dependency: r32-001-hostapd-Add-support-to-receive-AFC-power-update-comp.patch

Signed-off-by: Hariharan Basuthkar <quic_hbasuthk@quicinc.com>
---
 src/common/qca-vendor.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/common/qca-vendor.h b/src/common/qca-vendor.h
index 1113bdc..930643c 100644
--- a/src/common/qca-vendor.h
+++ b/src/common/qca-vendor.h
@@ -16320,6 +16320,9 @@ enum qca_wlan_vendor_attr_afc_event {
  * the attributes defined in enum qca_wlan_vendor_attr_afc_opclass_info and
  * enum qca_wlan_vendor_attr_afc_chan_eirp_info. Required attribute.
  *
+ * @QCA_WLAN_VENDOR_ATTR_AFC_RESP_HW_IDX: (u8) array index in wiphy @hw_chans
+ * to refer an underlying hw mac for which the AFC Response data is
+ * advertised.
  */
 enum qca_wlan_vendor_attr_afc_response {
 	QCA_WLAN_VENDOR_ATTR_AFC_RESP_INVALID = 0,
@@ -16331,6 +16334,7 @@ enum qca_wlan_vendor_attr_afc_response {
 	QCA_WLAN_VENDOR_ATTR_AFC_RESP_AFC_SERVER_RESP_CODE = 6,
 	QCA_WLAN_VENDOR_ATTR_AFC_RESP_FREQ_PSD_INFO = 7,
 	QCA_WLAN_VENDOR_ATTR_AFC_RESP_OPCLASS_CHAN_EIRP_INFO = 8,
+	QCA_WLAN_VENDOR_ATTR_AFC_RESP_HW_IDX = 9,
 
 	QCA_WLAN_VENDOR_ATTR_AFC_RESP_AFTER_LAST,
 	QCA_WLAN_VENDOR_ATTR_AFC_RESP_MAX =
-- 
2.34.1

